newIDE/app/src/UI/ImagesDisplay.js

```diff
@@ -0,0 +1,255 @@
+                  setMobileGridScrollX(event.currentTarget.scrollLeft)
+                }
+              >
+                {imagesUrls.map((url, index) => (
+                  <Grid
+                    item
+                    key={url}
+                    style={{
+                      scrollSnapAlign: horizontalOuterMarginToEatOnMobile
+                        ? 'center'
+                        : 'start',
+                    }}
+                  >
+                    <CardMedia>
+                      <CorsAwareImage
+                        src={url}
+                        style={{
+                          ...styles.mobileImageCarouselItem,
+                          height: mobileImageWidth / (16 / 9),
+                        }}
+                        alt={altTextTemplate.replace(
+                          /{imageIndex}/g,
+                          String(selectedImageIndex + 1)
+                        )}
+                      />
+                    </CardMedia>
+                  </Grid>
+                ))}
+              </Grid>
+            </div>
+          )}
+        </Measure>
+        <Line justifyContent="center">
+          <Text noMargin size="body2">
+            {currentlyViewedImageIndex + 1}/{imagesUrls.length}
+          </Text>
+        </Line>
+      </div>
+    );
+  }
+  return (
+    <Column noMargin>
+      <CorsAwareImage
+        style={styles.mainImage}
+        src={imagesUrls[selectedImageIndex]}
+        alt={altTextTemplate.replace(
+          /{imageIndex}/g,
+          String(selectedImageIndex + 1)
+        )}
+      />
+      <Line>
+        <Grid
+          classes={classesForGridContainer}
+          container
+          spacing={GRID_SPACING}
+          wrap="nowrap"
+          style={styles.grid}
+        >
+          {imagesUrls.map((url, index) => (
+            <Grid
+              item
+              key={url}
+              tabIndex={0}
+              onKeyPress={(
+                event: SyntheticKeyboardEvent<HTMLLIElement>
+              ): void => {
+                if (shouldValidate(event)) {
+                  setSelectedImageIndex(index);
+                }
+              }}
+            >
+              <CardMedia
+                onClick={() => setSelectedImageIndex(index)}
+                style={{
+                  ...styles.carouselItem,
+                  outline:
+                    index === selectedImageIndex
+                      ? 'solid 1px white'
+                      : undefined,
+                }}
+              >
+                <CorsAwareImage
+                  src={url}
+                  style={styles.imageCarouselItem}
+                  alt={altTextTemplate.replace(
+                    /{imageIndex}/g,
+                    String(index + 1)
+                  )}
+                />
+              </CardMedia>
+            </Grid>
+          ))}
+        </Grid>
+      </Line>
+    </Column>
+  );
+};
+
+export default ImagesDisplay;
```

Comment:
I'm not sure about the naming. It feels really... generic. But I don't have a better idea 😅 

`ResponsiveImagesRow` ?

Reply:
Could even be ScrollingImagesRow, to show that it's a "carousel of images".

Reply:
Actually, it's a gallery of images. ResponsiveImagesGallery?

Reply:
Good idea! I'll use that